lib/chef/http.rb

```diff
@@ -198,6 +197,10 @@ def streaming_request(path, headers={}, &block)
     def http_client(base_url=nil)
       base_url ||= url
       if chef_zero_uri?(base_url)
+        # PERFORMANCE CRITICAL: *MUST* lazy require here otherwise we load up webrick
+        # via chef-zero and that hits DNS (at *require* time) which may timeout,
+        # when for most knife/chef-client work we never need/want this loaded.
+        require 'chef/http/socketless_chef_zero_client'
```

Comment:
maybe add `unless defined?(SocketlessChefZeroClient)` here? I think the perf for a redundant require has improved in recent Ruby but I know it has been an issue in the past.